### DISCUSSION TITLE
Crazyradio range is abnormally short

### DISCUSSION THREAD

**Discussion by daffakanz** [bitcraze/discussions#1553] [Q&A]
Hello, I was trying to estimate geometry for setting up Lighthouse Positioning System. But soon I realize my Crazyradio range is shorter than what its supposed to be. I'm experiencing packet losses when my Crazyflie is away more than 1m from the Crazyradio.

This might help showing what I'm experiencing:

https://github.com/user-attachments/assets/a075edbf-f70e-4c2b-a99b-6652a34f9520

for higher resolution: [https://drive.google.com/file/d/1tkZs-LZP1_8Uz8Lb-SciUg4IZcJ7CUma/view?usp=sharing](url)

My testbed in lab is 4x4 meters room that clear from debris, So I'm sure my Crazyradio has clear line of sight to my Crazyflie. But in the lab there are also wifi modems, 3D printers, and other robotic items... Does this have any effect on my Crazyradio performance?

I tried to google and found this old thread [https://forum.bitcraze.io/viewtopic.php?t=4046](url) where someone is also having shorter range on their Crazyradio.
From one point of the answer there is 
> - Are you setting up big logging block? There might be a bug when using 32bytes packets that I am currently investigating. Limiting the size of the packets transmitted to the Crazyflie can help.

Can anyone show me how to do this in cfclient? or is there a new solution for this?

**Comment by tobbeanton** [comment]
There is something wrong, for sure. What firmware are you running? Do you get the same short range in bootloader mode? Check that the antenna on the crazyradio is attached well.

**Reply by daffakanz** [reply]
I'm running this on machine with specs:
- Ubuntu 22.04.4 LTS
- 8GB RAM
- Intel i3-10110U CPU

On the Crazyflie, I'm running "2024.2 - firmware-cf2-2024.2" firmware.

Unfortunately, yes I also have same issue in bootloader mode. I've checked the antenna few times even trying out other antennas (Crazyradio PA) that I have. I also have tried using Crazyradio 2.0, the issue still persist.

**Reply by tobbeanton** [reply]
I'm leaning towards a hardware problem. On the picture it looks like you have other crazyflies. Do they behave the same?

**Reply by daffakanz** [reply]
Which hardware do you mean here? The antennas or my machine?
Personally, I lean more to my machine's fault. Because the crazyradio 2.0 is brand new. The PA antennas were from old projects.
I will try to use my friend's laptop if the problem might be coming from my machine.

Other crazyflies also behave the same. They all have short range on my machine.

**Comment by daffakanz** [comment]
I've tested again today and other function is nice and behaving what it supposed to do as long as the Crazyflie is connected to the machine.
The range issue still persist even after using other firmware.
Minimal Range:
![IMG_20240919_114056_074](https://github.com/user-attachments/assets/48a5f140-9406-4666-81a6-c5745f5b1cef)
Maximal Range:
![IMG_20240919_114106_859](https://github.com/user-attachments/assets/cb30b367-eb67-4841-ad1f-df5b8b8bf940)

I succeeded in estimating the geometry for the system with my friend's help. The result has big error margin since I didn't do it properly. We had to move side by side with me holding my laptop and my friend holding the crazyflie.
Test flight:
[https://drive.google.com/file/d/1v9yHKwA-KO65KpPLQCmhrQYjgEayLYMd/view?usp=sharing](url)

**Reply by tobbeanton** [reply]
OK, so all crazyflies you have has this short range with the Crazyradio PA/2.0 you have? Can you try connecting with the Android or iPhone app to the Crazyflies and test the range?
